nomos-services/api/src/http/storage/mod.rs

```diff
@@ -0,0 +1,105 @@
+use std::{
+    collections::HashSet,
+    fmt::{Debug, Display},
+    hash::Hash,
+};
+
+use bytes::Bytes;
+use futures::Stream;
+use nomos_core::{
+    block::Block,
+    da::blob::{LightShare, Share},
+    header::HeaderId,
+};
+use nomos_storage::{
+    api::da::StorageDaApi,
+    backends::{rocksdb::RocksBackend, StorageBackend, StorageSerde},
+    StorageService,
+};
+use overwatch::services::{relay::OutboundRelay, AsServiceId, ServiceData};
+use serde::{de::DeserializeOwned, Serialize};
+
+pub mod adapters;
+
+#[async_trait::async_trait]
+pub trait StorageAdapter<StorageOp, RuntimeServiceId>
+where
+    StorageOp: StorageSerde + Send + Sync + 'static,
+{
+    type Backend: StorageBackend + Send + Sync + 'static;
+
+    async fn new(
+        storage_relay: OutboundRelay<
+            <StorageService<Self::Backend, RuntimeServiceId> as ServiceData>::Message,
+        >,
+    ) -> Self;
+
+    async fn get_light_share<DaShare>(
+        storage_relay: OutboundRelay<
+            <StorageService<RocksBackend<StorageOp>, RuntimeServiceId> as ServiceData>::Message,
+        >,
+        blob_id: <DaShare as Share>::BlobId,
+        share_idx: <DaShare as Share>::ShareIndex,
+    ) -> Result<Option<<DaShare as Share>::LightShare>, super::DynError>
+    where
```

Comment:
Why we need these constraints on the trait definition, wouldn't it be enough for the implementation to have these constraints?

Reply:
Okay, tried with the associated types, the results would be similar, I'm ok with this solution.